.github/workflows/docs-workflow.yaml

```diff
@@ -0,0 +1,72 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2023-2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+name: Docs generation for Github Pages
+
+on:
+  push:
+    paths:
```

Comment:
This workflow is triggered when changes have been committed in the `docs`  folder or in the `docs-workflow.yaml` file.
However, most of the changes in the generated doxygen will only be in the source files, right?

Reply:
You are right. I will remove paths and leave action to trigger on source changed on master and release branches.